### PR TITLE
feat: Monitoring assistant

### PR DESCRIPTION
**Pull request by michaelwittig**

### CODE REVIEW

File: marbot-aws-account-connection.yml [hunk 1]
```diff
@@ -139,6 +146,83 @@ Resources:
...
+          - 'elasticloadbalancing:DescribeLoadBalancers'
+          - 'elasticloadbalancing:DescribeTargetGroups'
+          - 'events:DeleteRule'
+          - 'events:DescribeRule'
+          - 'events:ListTagsForResource'
+          - 'events:ListTargetsByRule'
+          - 'events:PutRule'
+          - 'events:PutTargets'
+          - 'events:RemoveTargets'
+          - 'iam:CreateRole'
```
Comment (andreaswittig) [inline marbot-aws-account-connection.yml:178]: Should we use permission boundaries to mitigate the risk of granting full administrator access?

File: marbot-aws-account-connection.yml [hunk 2]
```diff
@@ -139,6 +146,83 @@ Resources:
...
+          - 'iam:DeleteRole'
+          - 'iam:DeleteRolePolicy'
+          - 'iam:GetRole'
+          - 'iam:GetRolePolicy'
+          - 'iam:ListAttachedRolePolicies'
+          - 'iam:ListInstanceProfilesForRole'
+          - 'iam:ListRolePolicies'
+          - 'iam:PassRole'
```
Comment (andreaswittig) [inline marbot-aws-account-connection.yml:186]: Could we restrict that to certain roles?

File: marbot-aws-account-connection.yml [hunk 3]
```diff
@@ -139,6 +146,83 @@ Resources:
...
+          - 'iam:PutRolePolicy'
+          - 'iam:UpdateRoleDescription'
+          - 'lambda:AddPermission'
```
Comment (andreaswittig) [inline marbot-aws-account-connection.yml:189]: Could we restrict the `lambda:` permissions to functions with a specific name prefix?

File: marbot-aws-account-connection.yml [hunk 4]
```diff
@@ -139,6 +146,83 @@ Resources:
           Condition:
             StringEquals:
               'sts:ExternalId': !Select [2, !Split ['/', !Ref 'AWS::StackId']]
+  MonitoringAssistantPolicy:
+    Condition: MonitoringAssistantEnabled
+    Type: 'AWS::IAM::Policy'
+    Properties:
+      PolicyDocument:
```
Comment (andreaswittig) [inline marbot-aws-account-connection.yml:153]: Overall, I'm wondering wether we should restrict access to resources based on a name prefix. What do you thing, @michaelwittig?

**Review by andreaswittig — CHANGES_REQUESTED** [review]

File: marbot-aws-account-connection.yml (hunk 3 above)
Comment (michaelwittig) [reply marbot-aws-account-connection.yml]: done

File: marbot-aws-account-connection.yml (hunk 4 above)
Comment (michaelwittig) [reply marbot-aws-account-connection.yml]: done

File: marbot-aws-account-connection.yml (hunk 2 above)
Comment (michaelwittig) [reply marbot-aws-account-connection.yml]: done

File: marbot-aws-account-connection.yml (hunk 1 above)
Comment (michaelwittig) [reply marbot-aws-account-connection.yml]: But how do we deploy the permission boundary? If you trust us creating the permission boundary role you can also trust us creating the roles itself?

Comment (andreaswittig) [reply marbot-aws-account-connection.yml]: In theory, the customer is able to review this template before creating the stack. So, it is possible to review the permission boundary. Same is true for updating the stack.

But when marbot uses the IAM role to deploy the monitoring configuration the customer has no chance to review the IAM changes.

Comment (michaelwittig) [reply marbot-aws-account-connection.yml]: I see

**Pull request merged**